integration_tests/src/main/python/cmp_test.py

```diff
@@ -31,7 +31,7 @@ def test_eq(data_gen):
                 s2 == f.col('b'),
                 f.lit(None).cast(data_type) == f.col('a'),
                 f.col('b') == f.lit(None).cast(data_type),
-                f.col('a') == f.col('b')))
+                f.col('a') == f.col('b')), conf={'spark.sql.legacy.allowNegativeScaleOfDecimal': 'true'})
```

Comment:
If this is going to be common for a lot of decimal tests can we make it something in data_gen that can be used by multiple tests?

Reply:
Done